packages/core/src/baseCogniteClient.ts

```diff
@@ -58,14 +64,32 @@ export interface ApiKeyLoginOptions extends Project {
   apiKey: string;
 }
 
-export const AAD_OAUTH = 'AAD_OAUTH';
-export const CDF_OAUTH = 'CDF_OAUTH';
-export const ADFS_OAUTH = 'ADFS_OAUTH';
-export type AuthFlowType =
-  | typeof AAD_OAUTH
-  | typeof CDF_OAUTH
-  | typeof ADFS_OAUTH;
+export type AAD_OAUTH = {
+  type: 'AAD_OAUTH';
+  options: OAuthLoginForAADOptions;
+};
+export type CDF_OAUTH = {
+  type: 'CDF_OAUTH';
+  options: OAuthLoginForCogniteOptions;
+};
+export type ADFS_OAUTH = {
+  type: 'ADFS_OAUTH';
+  options: OAuthLoginForADFSOptions;
+};
+export type OIDC_AUTHORIZATION_CODE_FLOW = {
+  type: 'OIDC_AUTHORIZATION_CODE_FLOW';
+  options: OAuthLoginForOIDCAuthFlowOptions;
+};
 
+export type AuthFlowType =
```

Comment:
so glad we did this. 🙌 much more readable and only by using a basic union with common fields

Reply:
Agree — this approach looks great